src/shared/lib/comparison/ComparisonStore.ts

```diff
@@ -2170,6 +2153,107 @@ export default abstract class ComparisonStore {
         });
     }
 
+    readonly molecularProfilesInStudies = remoteData<MolecularProfile[]>(
+        {
+            await: () => [this.studyIds],
+            invoke: async () => {
+                return client.fetchMolecularProfilesUsingPOST({
+                    molecularProfileFilter: {
+                        studyIds: this.studyIds.result,
+                    } as MolecularProfileFilter,
+                });
+            },
+        },
+        []
+    );
+
+    readonly customDriverAnnotationProfiles = remoteData<MolecularProfile[]>(
+        {
+            await: () => [this.molecularProfilesInStudies],
+            invoke: async () => {
+                return _.filter(
+                    this.molecularProfilesInStudies.result,
+                    (molecularProfile: MolecularProfile) =>
+                        // discrete CNA's
+                        (molecularProfile.molecularAlterationType ===
+                            AlterationTypeConstants.COPY_NUMBER_ALTERATION &&
+                            molecularProfile.datatype ===
+                                DataTypeConstants.DISCRETE) ||
+                        // mutations
+                        molecularProfile.molecularAlterationType ===
+                            AlterationTypeConstants.MUTATION_EXTENDED ||
+                        // structural variants
+                        molecularProfile.molecularAlterationType ===
+                            AlterationTypeConstants.STRUCTURAL_VARIANT
+                );
+            },
+        },
+        []
+    );
+
+    readonly customDriverAnnotationReport = remoteData<IDriverAnnotationReport>(
+        {
+            await: () => [this.customDriverAnnotationProfiles],
+            invoke: async () => {
```

Comment:
same (use promise.resolve)

Reply:
Can't do it here since there is an _await_ in the body (at least the compiler notifies me of this).